core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.kt

```diff
@@ -359,16 +360,25 @@ class AddNoteDialog : DialogFragment() {
     val noteFile =
       File(notesFolder.absolutePath, "$articleNoteFileName.txt")
     val noteDeleted = noteFile.delete()
+    val noteText = dialogNoteAddNoteBinding?.addNoteEditText?.text.toString()
     if (noteDeleted) {
       dialogNoteAddNoteBinding?.addNoteEditText?.text?.clear()
       mainRepositoryActions.deleteNote(articleNoteFileName)
       disableMenuItems()
-      context.toast(R.string.note_delete_successful, Toast.LENGTH_LONG)
+      view?.snack(
+        R.string.note_delete_successful,
+        R.string.undo,
+        actionClick = { onUndoCLicked(noteText) }
+      )
     } else {
       context.toast(R.string.note_delete_unsuccessful, Toast.LENGTH_LONG)
     }
   }
 
+  private fun onUndoCLicked(text: String) {
```

Comment:
We should name this function more accurately, ```restoreDeletedNote()``` would be better.

Reply:
Okay then i am changing the name from onUndoClicked() -> restoreDeletedNote()